projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1073,12 +1073,29 @@ function get_paywall_blocks( $newsletter_access_level ) {
 		? __( 'Already a paid subscriber?', 'jetpack' )
 		: __( 'Already a subscriber?', 'jetpack' );
 
-	$sign_in = '';
-	if ( ! is_user_logged_in() && ( new Host() )->is_wpcom_simple() ) {
-		$sign_in_link = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
+	$sign_in             = '';
+	$show_button_sign_in = false;
+	$host                = new Host();
+	if ( ! is_user_logged_in() && $host->is_wpcom_simple() ) {
+		$show_button_sign_in = true;
+		$sign_in_link        = wpcom_logmein_redirect_url( get_current_url(), false, null, 'link' );
+		$button_text         = esc_html__( 'Log in', 'jetpack' );
+	} elseif ( class_exists( 'Automattic\Jetpack\Connection\Tokens\Jetpack_Token_Subscription_Service' ) &&
+		! $host->is_wpcom_platform() ) {
+		// We are on Jetpack and no cookie is set
+		$show_button_sign_in = true;
+		$sign_in_link        = '';
+		$access_question     = __( 'I am already subscribed.', 'jetpack' );
+		if ( ! Jetpack_Token_Subscription_Service::has_token_from_cookie() ) {
+			$button_text = esc_html__( 'Retrieve subscriptions', 'jetpack' );
```

Comment:
We can keep the "log in" terminology here for clarity

Reply:
I like not using "log in", specially with this approach where we are not really login the user, in particular for Jetpack sites. "Get access" maybe?

Reply:
I get where you both are coming with this (I talked about it with Fab in Malaga), but I don't think it really matters or that there would be confusion; it's right in the context of paywall, and they know they are subscriber, so I don't think we'll have issues. 

Let's not "optimize for minority with expense of clarity for majority". :-) There will always be very few admins of the site (who might expect to login to site admin) vs very many subscribers/readers. The few admins will learn quickly they can't use this to actually login.

"Sign in/log in" is very established wording too — and thus easy to translate in every language. Anything else will quickly get fuzzier quickly, especially across translations.